lib/viewmodels/onboarding_model.dart

```diff
@@ -0,0 +1,43 @@
+import 'package:app/locator.dart';
+import 'package:app/routes.dart';
+import 'package:app/services/auth.dart';
+import 'package:app/services/dynamicLinks.dart';
+import 'package:app/services/navigation_service.dart';
+import 'package:app/services/pushNotifications.dart';
+import 'package:app/services/remote_config_service.dart';
+import 'package:app/viewmodels/base_model.dart';
+import 'package:firebase_core/firebase_core.dart';
+
+class OnBoardingViewModel extends BaseModel {
+  final AuthenticationService _authenticationService =
+      locator<AuthenticationService>();
+  final NavigationService _navigationService = locator<NavigationService>();
+
+  Future handleStartUpLogic() async {
```

Comment:
Does this need to be here? Can this stay in the start up viewmodel and then that will handle navigating to this page?

Reply:
Yes this is something that I did when I was having a chain of thoughts and should've deleted from here later on. I will clean things up and leave handling that to the start up viewmodel.

Reply:
> @ahmedtilal this looks great thank you so much for this!
> 
> I think my main comment is the repeated code between all the pages. I like how the pages are flexible so that the `FirstOnBoardingPage` can be handled differently to the rest but I think for pagese 2 -4 we should have a single base class that takes in the text and image as arguments. Hopefully, this will reduce the amount of code required. Let me know what you think?

Yes that's a good idea. We sure can do such refactoring. I'll let you know once I've done all these changes.